packages/@aws-cdk/aws-lambda/lib/code.ts

```diff
@@ -46,20 +47,20 @@ export abstract class Code {
   public abstract readonly isInline: boolean;
 
   /**
-   * Called during stack synthesis to render the CodePropery for the
-   * Lambda function.
+   * Called when the lambda or layer is initialized to allow this object to bind
+   * to the stack, add resources and have fun.
    *
-   * @param resource the resource to which the code will be attached (a CfnFunction, or a CfnLayerVersion).
-   *
-   * @internal
+   * @param scope The binding scope. Don't be smart about trying to down-cast or
+   * assume it's initialized. You may just use it as a construct scope.
    */
-  public abstract _toJSON(resource?: cdk.CfnResource): CfnFunction.CodeProperty;
+  public abstract bind(scope: cdk.Construct): CfnFunction.CodeProperty;
```

Comment:
Don't think this should be returning an L1.

Reply:
Fixed